src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunnerPool.cs

```diff
@@ -110,12 +112,21 @@ private VsTestRunner TakeRunner()
                     Monitor.Wait(_lck);
                 }
             }
-
+            // should not happen, placed to chase an elusive bug on MacOS
+            if (runner == null)
```

Comment:
We could also check whether the VsTestRunner is actually still connected with VsTest because I believe the VsTestConsole exposes this information. If we detect the VsTestConsole session has ended for some reason in TakeRunner we could delete this instance and create a new one. In case vstest crashes or something we don't end up supplying the testrun with an invalid test agent.